letter-shop.md

```diff
@@ -0,0 +1,232 @@
+# Letter Shop Tutorial
+
+## What you need before you start:
+
+* a laptop with an internet connection and NodeJS installed
+* basic knowledge of the command line terminal
+* basic knowledge of JavaScript, including Promises and Buffers
+* (optional) the concept of one-way hash functions
+
+## What you'll learn:
+
+* the concept Hash Time Lock Agreements (HTLAs)
+* the Ledger Plugin Interface (LPI)
+* how to build a paid web app using Interledger!
+
+## Step 1:
+
+Save the following JavaScript as `shop.js`:
+
+```js
+const http = require('http')
+const crypto = require('crypto')
+const Plugin = require('ilp-plugin-xrp-escrow')
+function base64 (buf) { return buf.toString('base64').replace(/\+/g, '-').replace(/\//g, '_').replace(/=+$/, '') }
+
+let fulfillments = {}
+let letters = {}
+
+const plugin = new Plugin({
+  secret: 'ssGjGT4sz4rp2xahcDj87P71rTYXo',
+  account: 'rrhnXcox5bEmZfJCHzPxajUtwdt772zrCW',
+  server: 'wss://s.altnet.rippletest.net:51233',
+  prefix: 'test.crypto.xrp.'
+})
+
+plugin.connect().then(function () {
+  plugin.on('incoming_prepare', function (transfer) {
+    plugin.fulfillCondition(transfer.id, fulfillments[transfer.executionCondition]).catch(function () {})
```

Comment:
This is an insecure way to handle fulfilling payments (because you're not checking the amount). We should not encourage people to do this themselves because they are likely to make mistakes like this and lose money. The example should use the `ilp` module instead. If neither PSK nor IPR is suitable for this we should discuss why they aren't, come up with an alternative, and include it in that module.

Reply:
I'll check the amount (the second tutorial does check the transfer amount against the IPP amount). Checking the expiry is already done by the ledger. I don't think we need to include a module for that?

The payment request here is a human-readable phrase. Further on, the proxy actually checks if the first word on the page is 'Please', and only pays if it is. :) I'm starting with this simplistic flow on purpose; using IPR would just hide the under-the-hood semantics from sight.

I agree about setting a price per letter though, and then checking that. It's currently a 'pay what you want' shop and that's just silly. :)